src/fingerprint.cpp

```diff
@@ -27,12 +27,33 @@ std::string uri_fingerprint(const std::string& uri) {
 
     std::stringstream fingerprint;
 
-    Poco::URI uri_parsed(uri);
-    const std::string& path = uri_parsed.getPath();
+    // faup_options_new() is not thread safe, and should only be runned once per
+    // code, it is also the part that loads the cached publicsuffix.org file
+    faup_options_t* faup_opts;
+    faup_opts = faup_options_new();
+
+    // no need for the default csv output
+    faup_opts->output = FAUP_OUTPUT_NONE;
+
+    // modules slow down faup_init(), should be loaded only when needed
+    faup_opts->exec_modules = FAUP_MODULES_NOEXEC;
+
+    // fh is a pointer to a c struct that has, among others, all the positions
+    // where the uri splits in host, tld, etc.
+    faup_handler_t* fh;
```

Comment:
This pointer should be freed before the end of the method, faup offers a dedicated method `faup_terminate`
- https://github.com/stricaud/faup/blob/master/doc/library.md
- https://cpp.hotexamples.com/fr/examples/-/-/faup_init/cpp-faup_init-function-examples.html